man/fi_profile.3.md

```diff
@@ -0,0 +1,260 @@
+interface. Providers define events and use them to report an operation 
+occured or a state changed where one or more variables may change their 
+values. Users can register callback functions to one or more events, and 
+receive notifications from providers once an event occurs.
+
+In order to support the profiling interface, an endpoing or a domain in a 
+provider needs to maintain a fid_profile object, and implement the profiling 
+operations there. To starts a profiling,  an open call must be made on the 
+target endpoint or domain object to obtain the fid_profile object.  If the 
+target object does not support profiling, the open call will fail with error 
+code -FI_ENOSYS. Once a fid_profile object is obtained successfully, it can be 
+used to query available variables or events, and read variables' values.
+
+# VARIABLES
+
+The profiling variables are defined in providers, and become accessable 
+(readable) externally through the profiling interface.  A variable can be 
+a configuration setting, a runtime state or a counter in an endpoint or a 
+domain in a provider. Some variables are common and available in most 
+providers, while others are provider-specific, and are only available in 
+one provider. Users can query available variables and their values using 
+the profiling interface.
+
+FI_VAR_UNEXP_MSG_CNT is a common variable and it counts the number of  
+unexpected message recevied and unmatched. The variable has a unint64_t 
+data value.
+
+A variable is specified using struct fi_profile_desc and is described below.
+
+# EVENTS
+
+The profiling event is defined by the providers. It is used to notify users 
+that an operation has occured or a state has been moved, which may change 
+the values of some variables. Event is a way for providers to inform changes 
+on the variables. Similar to the variables, some events are pre-defined and 
+shared by many providers, while others are provider-specific events and 
+defined by each provider. Users can register a callback function on an event 
+to get notified when the event occurs.
+
+FI_EVENT_UNEXP_MSG_RECVD and FI_EVENT_UNEXP_MSG_MATCHED are common events 
+which imply there are changes on variable FI_VAR_UNEXP_MSG_CNT.
+
+Same as variables, events are specified by the fi_profile_desc struct.  
+
+{% highlight c %}
+struct fi_profile_desc {
+	uint32_t id;
+	enum fi_datatype type;
+	size_t size;
+	const char *name;
+	const char *desc;
+};
+{% endhighlight %}
+
+## id
+
+Each variable or event has a unique identifier (id). Providers provide this 
+id and users use it to get the variable value or register callbacks for 
+the event.
+
+## type
+
+The data type of the variable. 
+
+## size
+
+The size (in byte) of the data. "type" and "size" can be used to support 
+any provider-specific data.
+
+## name
+
+Reference to the name of a variable or an event. It is either defined or 
+allocated by providers. Variables or events with the same name should have 
```

Comment:
Wording: "Reference to the name of a variable or an event.  Storage is allocated and managed by the provider."

Similar wording for desc below.